### DISCUSSION TITLE
Secondary store limit

### DISCUSSION THREAD

**Discussion by 0xshinobii** [aptos-labs/aptos-developer-discussions#326] [Questions]
### Describe your question in detail.

Hey folks, is there any limit on several secondary stores the module can create, and also how do we estimate gas for creating and maintaining a store? Is it advisable to have multiple secondary stores per user to maintain funds deposited by user or is there a better way to do it in move? The behavior in solidity that I want to achieve is maintaining multiple credit accounts (contracts) per user which hold user funds but are in control of the protocol.

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by 0x-j** [comment]
I think it makes sense to have a secondary store for each credit account. I don't think there's a hard limit on the number because the store owner doesn't have a list to track all the stores (unless you save this info yourself), so even if you have a lot it won't affect storage or performance. 

In general, state on aptos is generally stored as distributed as possible.

**Reply by 0x-j** [reply]
We have some instructions on how to do gas profiling. 

https://aptos.dev/en/build/cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling#gas-profiling
